### PR TITLE
Criação do pre autorize para download de arquivo.

### PR DESCRIPTION
**Pull request by jacquesnier**
Para revisor e autor do trabalho e organizador do evento.

### CODE REVIEW

File: aplicacao/contest/src/main/java/ufc/quixada/npi/contest/config/security/CustomMethodSecurityExpressionRoot.java [hunk 1]
```diff
@@ -10,14 +10,10 @@
 import ufc.quixada.npi.contest.model.ParticipacaoTrabalho;
 import ufc.quixada.npi.contest.model.Pessoa;
 import ufc.quixada.npi.contest.model.Secao;
-import ufc.quixada.npi.contest.model.Trabalho;
 import ufc.quixada.npi.contest.service.TrabalhoService;
```
Comment (npiuser) [inline aplicacao/contest/src/main/java/ufc/quixada/npi/contest/config/security/CustomMethodSecurityExpressionRoot.java]: MINOR Remove this unused import 'ufc.quixada.npi.contest.service.TrabalhoService'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: aplicacao/contest/src/main/java/ufc/quixada/npi/contest/config/security/CustomMethodSecurityExpressionRoot.java [hunk 2]
```diff
@@ -142,16 +138,8 @@ public boolean isOrientadorInTrabalho(Long trabalhoId){
...
-    	for(ParticipacaoTrabalho participacao : pessoa.getParticipacoesTrabalho()){
-    		if(participacao.getTrabalho().getId() == trabalhoId && (participacao.getPapel()== Tipo.COAUTOR || participacao.getPapel()== Tipo.REVISOR || participacao.getPapel()== Tipo.AUTOR)) return true;
-    	}
-    	Trabalho trabalho = trabalhoService.getTrabalhoById(trabalhoId);
-    	
-    	for(ParticipacaoEvento participacao : pessoa.getParticipacoesEvento()){
-    		if(participacao.getEvento() == trabalho.getEvento() && participacao.getPapel() == Tipo.ORGANIZADOR) return true;
-    	}
+    public boolean isResponsavelInTrabalho(Long trabalhoId, Long eventoId){
+    	if (isAutorInTrabalho(trabalhoId) || isOrientadorInTrabalho(trabalhoId) || isCoautorInTrabalho(trabalhoId) || isRevisorInTrabalho(trabalhoId) || isOrganizadorInEvento(eventoId)) return true;
```
Comment (npiuser) [inline aplicacao/contest/src/main/java/ufc/quixada/npi/contest/config/security/CustomMethodSecurityExpressionRoot.java]: MINOR Replace this if-then-else statement by a single return statement. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1126)

**Comment by npiuser** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [CustomMethodSecurityExpressionRoot.java#L3](https://github.com/npi-ufc-qxd/contest/blob/7f219d1b8b868580439baff42e3ffb1dd49e133f/aplicacao/contest/src/main/java/ufc/quixada/npi/contest/config/security/CustomMethodSecurityExpressionRoot.java#L3): Remove this unused import 'org.springframework.beans.factory.annotation.Autowired'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by npiuser** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [CustomMethodSecurityExpressionRoot.java#L3](https://github.com/npi-ufc-qxd/contest/blob/37173c4e987ab16a78b07e4011a4a3e5bb3ffb83/aplicacao/contest/src/main/java/ufc/quixada/npi/contest/config/security/CustomMethodSecurityExpressionRoot.java#L3): Remove this unused import 'org.springframework.beans.factory.annotation.Autowired'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: aplicacao/contest/src/main/java/ufc/quixada/npi/contest/config/security/CustomMethodSecurityExpressionRoot.java [hunk 3]
```diff
@@ -142,16 +136,8 @@ public boolean isOrientadorInTrabalho(Long trabalhoId){
...
-    	for(ParticipacaoTrabalho participacao : pessoa.getParticipacoesTrabalho()){
-    		if(participacao.getTrabalho().getId() == trabalhoId && (participacao.getPapel()== Tipo.COAUTOR || participacao.getPapel()== Tipo.REVISOR || participacao.getPapel()== Tipo.AUTOR)) return true;
-    	}
-    	Trabalho trabalho = trabalhoService.getTrabalhoById(trabalhoId);
-    	
-    	for(ParticipacaoEvento participacao : pessoa.getParticipacoesEvento()){
-    		if(participacao.getEvento() == trabalho.getEvento() && participacao.getPapel() == Tipo.ORGANIZADOR) return true;
-    	}
+    public boolean isResponsavelInTrabalho(Long trabalhoId, Long eventoId){
+    	if (isAutorInTrabalho(trabalhoId) || isCoautorInTrabalho(trabalhoId) || isRevisorInTrabalho(trabalhoId) || isOrganizadorInEvento(eventoId)) return true;
```
Comment (npiuser) [inline aplicacao/contest/src/main/java/ufc/quixada/npi/contest/config/security/CustomMethodSecurityExpressionRoot.java]: MINOR Replace this if-then-else statement by a single return statement. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1126)

**Comment by npiuser** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: aplicacao/contest/src/main/java/ufc/quixada/npi/contest/config/security/CustomMethodSecurityExpressionRoot.java [hunk 4]
```diff
@@ -142,16 +136,8 @@ public boolean isOrientadorInTrabalho(Long trabalhoId){
...
-    	for(ParticipacaoTrabalho participacao : pessoa.getParticipacoesTrabalho()){
-    		if(participacao.getTrabalho().getId() == trabalhoId && (participacao.getPapel()== Tipo.COAUTOR || participacao.getPapel()== Tipo.REVISOR || participacao.getPapel()== Tipo.AUTOR)) return true;
-    	}
-    	Trabalho trabalho = trabalhoService.getTrabalhoById(trabalhoId);
-    	
-    	for(ParticipacaoEvento participacao : pessoa.getParticipacoesEvento()){
-    		if(participacao.getEvento() == trabalho.getEvento() && participacao.getPapel() == Tipo.ORGANIZADOR) return true;
-    	}
+    public boolean isResponsavelInTrabalho(Long trabalhoId, Long eventoId){
+    	if (isAutorInTrabalho(trabalhoId) || isOrientadorInTrabalho(trabalhoId) || isCoautorInTrabalho(trabalhoId) || isRevisorInTrabalho(trabalhoId) || isOrganizadorInEvento(eventoId)) return true;
```
Comment (npiuser) [inline aplicacao/contest/src/main/java/ufc/quixada/npi/contest/config/security/CustomMethodSecurityExpressionRoot.java]: MINOR Replace this if-then-else statement by a single return statement. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1126)

**Comment by npiuser** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by jacquesnier** [comment]
@zarathon , eu tinha acabado de fazer outro commit, é uma modificação besta, mas vc aceitou sem passar pelos testes... tanto é que seu merge tá com o x lá...

**Pull request merged**